src/wasm-interpreter.h

```diff
@@ -3836,10 +3836,14 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     auto fail = Literal::makeFromInt64(-1, memory->addressType);
     Flow ret = Literal::makeFromInt64(memorySize, addressType);
     uint64_t delta = flow.getSingleValue().getUnsigned();
-    if (delta > uint32_t(-1) / Memory::kPageSize && addressType == Type::i32) {
+    uint64_t maxAddr =
+      addressType == Type::i32 ? uint64_t(uint32_t(-1)) : uint64_t(-1);
```

Comment:
Use `UINT64_MAX` and `UINT32_MAX` here?

Reply:
Yuck, that is so much hard to read than `UINT32_MAX`, no?

Reply:
Heh, I don't feel strongly. I guess it is more C++ey and can support all types and all bounds. But I wouldn't object to switching.

Reply:
I get that binaryen is fully in on all the C++ things :)   So lgtm either way, I just wish C++ wasn't so verbose sometimes.